src/main/java/com/box/sdk/BoxGroupMembership.java

```diff
@@ -158,20 +163,41 @@ public Info(String json) {
         /**
          * Gets the level of access the user has.
          * @return the level of access the user has.
+         * @deprecated use getMembershipRole instead.
          */
+        @Deprecated
         public Role getRole() {
             return this.role;
         }
 
         /**
          * Sets the level of access the user has.
          * @param role the new level of access to give the user.
+         * @deprecated use setMembershipRole instead.
          */
+        @Deprecated
         public void setRole(Role role) {
             this.role = role;
             this.addPendingChange("role", role.toJSONString());
         }
 
+        /**
+         * Gets the level of access the user has.
+         * @return the level of access the user has.
+         */
+        public MembershipRole getMembershipRole() {
```

Comment:
```suggestion
        public MembershipRole getRole() {
```
This is less of a change, but does it really make things that much easier for a client adopting the change if we preserve the name of the deprecated method?

Reply:
We can't preserve the getRole name in the new implementation. Java doesn't allow that even if they have a different return type. They would need to have different name and/or parameters.